assets/charts/control-plane/kube-apiserver/templates/kube-apiserver.yaml

```diff
@@ -61,6 +33,22 @@ spec:
         checkpointer.alpha.coreos.com/checkpoint: "true"
         seccomp.security.alpha.kubernetes.io/pod: 'docker/default'
     spec:
+      affinity:
+        podAntiAffinity:
+          # Have a soft policy so that two apiservers will run on single controller node. This can
```

Comment:
This does not work well, as while creating a cluster with multiple controllers, you will end up with all instances of kube-apiserver to be scheduled on a first controller node which shows up, which may overload it and won't provide HA. We currently have no way to re-balance them once new controller nodes joins the cluster. This is why we use DaemonSet instead.

Reply:
Add DS for all the control plane nodes. Since the problem is found in all of them.